service/src/main/java/bio/terra/cbas/service/RunSetsService.java

```diff
@@ -253,6 +263,20 @@ private WdsRecordResponseDetails fetchWdsRecords(
         recordIdsWithError.put(recordId, e.getMessage());
       }
     }
+    micrometerMetrics.stopTimer(
+        wdsFetchRecordsSample,
+        "wds_fetch_records_timer",
+        "run_set_id",
+        runSet.runSetId().toString(),
+        "total_records_requested",
+        "%s".formatted(request.getWdsRecords().getRecordIds().size()),
+        "failed_record_requests",
+        "%s".formatted(recordIdsWithError.size()),
+        "failure_rate",
+        "%s"
+            .formatted(
+                (double) recordIdsWithError.size()
+                    / request.getWdsRecords().getRecordIds().size()));
```

Comment:
TOL: I wonder if it would be better to maybe just record `failure_percentage` or `failure_rate`? We plan to already emit number of record IDs in request in separate metric and `failed_record_requests` is kind of duplicate to `failure_rate`. What do you think?

Reply:
I don't feel that strongly about it. It doesn't take up much space, and including it here might make browsing the logs slightly easier for someone, but you're right that it's not hard to connect the dots across two metrics. 

My vote is to keep it here, but I'll defer to you if you want it removed

Reply:
I don't see any harm in keeping it in customDimensions. If it becomes an issue we can remove it 👍

Reply:
I had to resolve a merge conflict, and in doing so I removed `total_records_requested` (because it's redundant) but I kept `failed_record_requests` because I think it will be convenient for the reasons I stated above (and it's measuring something slightly different than the failure rate)